### PR TITLE
PR for Issue 25524: Fix x5t calculation in OIDC Private Key JWT

### LINKED ISSUE #25524: Fix x5t calculation for OIDC Private Key JWT
The `x5t` value for the JWT should be the thumbprint of the public key that corresponds to the private key used to sign the JWT. I believe the current `x5t` calculation in https://github.com/OpenLiberty/open-liberty/blob/integration/dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/auth/PrivateKeyJwtAuthMethod.java#L89 is incorrect and needs to be fixed.

### PR DESCRIPTION
**Pull request by ayoho**
Updates the code to look for the public key that corresponds to the private key that is used to sign the JWT for client authentication. The `x5t` value is calculated using the public key and is included in the JWT that's created.
    
Resolves #25524

### CODE REVIEW

**Comment by ayoho** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3

**Comment by ayoho** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3

**Comment by ayoho** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3

**Comment by ayoho** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3

File: dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/auth/PrivateKeyJwtAuthMethod.java [hunk 1]
```diff
@@ -81,18 +160,39 @@ private String getSignedJwt(JwtClaims claims) throws Exception {
...
         try {
-            String x5t = new String(Base64.getUrlEncoder().encode(MessageDigest.getInstance("SHA-1").digest(clientAssertionSigningKey.getEncoded())));
-            jws.setHeader("x5t", x5t);
-        } catch (NoSuchAlgorithmException e) {
-            // SHA-1 should be supported; should be able to ignore
-            if (tc.isDebugEnabled()) {
-                Tr.debug(tc, "Caught exception getting x5t for signing key: " + e);
-            }
+            X509Certificate x509Cert = keyStoreService.getX509CertificateFromKeyStore(storeRef, keyAliasName);
+            return X509Util.x5t(x509Cert);
```
Comment (arunavemulapalli) [inline dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/auth/PrivateKeyJwtAuthMethod.java:191]: looks good

**Review by arunavemulapalli — APPROVED** [review]
I just focused on x5t change. lgtm. 
I am assuming other changes mostly related to refactoring or/and to get to the necessary ssl/keystore services and utilities.

**Comment by posmith** [comment]
L2 message review completed

**Review by helyarp — COMMENTED** [review]
Looks good!

**Pull request merged**